File: nan21.dnet.module.hr/nan21.dnet.module.hr.ui.extjs/src/main/webapp/src/ds/PositionRequirementDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.hr.job.ds.model.PositionRequirementDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"version", type:"int", useNull:true},
		{name:"entityFQN", type:"string"},
		{name:"positionId", type:"int", useNull:true},
		{name:"requirementId", type:"int", useNull:true},
		{name:"requirement", type:"string"},
		{name:"notes", type:"string"}
	]
});

Ext.define("net.nan21.dnet.module.hr.job.ds.model.PositionRequirementDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"positionId", type:"int", useNull:true},
		{name:"positionId_From",type:"int", useNull:true},
		{name:"positionId_To",type:"int", useNull:true},
		{name:"requirementId", type:"int", useNull:true},
		{name:"requirementId_From",type:"int", useNull:true},
		{name:"requirementId_To",type:"int", useNull:true},
		{name:"requirement", type:"string"},
		{name:"notes", type:"string"}
	] 
});
	
Ext.define("net.nan21.dnet.module.hr.job.ds.param.PositionRequirementDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 

File: nan21.dnet.module.hr/nan21.dnet.module.hr.presenter/src/main/java/net/nan21/dnet/module/hr/job/ds/model/PositionRequirementDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2010 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.hr.job.ds.model;

import net.nan21.dnet.core.presenter.model.base.AbstractAuditableDs;

import net.nan21.dnet.module.hr.job.domain.entity.PositionRequirement;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = PositionRequirement.class)
public class PositionRequirementDs extends
        AbstractAuditableDs<PositionRequirement> {

    public static final String fPOSITIONID = "positionId";
    public static final String fREQUIREMENTID = "requirementId";
    public static final String fREQUIREMENT = "requirement";
    public static final String fNOTES = "notes";

    @DsField(join = "left", path = "position.id")
    private Long positionId;

    @DsField(join = "left", path = "requirement.id")
    private Long requirementId;

    @DsField(join = "left", path = "requirement.name")
    private String requirement;

    @DsField()
    private String notes;

    public PositionRequirementDs() {
        super();
    }

    public PositionRequirementDs(PositionRequirement e) {
        super(e);
    }

    public Long getPositionId() {
        return this.positionId;
    }

    public void setPositionId(Long positionId) {
        this.positionId = positionId;
    }

    public Long getRequirementId() {
        return this.requirementId;
    }

    public void setRequirementId(Long requirementId) {
        this.requirementId = requirementId;
    }

    public String getRequirement() {
        return this.requirement;
    }

    public void setRequirement(String requirement) {
        this.requirement = requirement;
    }

    public String getNotes() {
        return this.notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

}
